### ISSUE TITLE
Assigning between sets of dataframe columns throws ValueError

### ISSUE THREAD

**Issue by JnsLns** [modin-project/modin#3258]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Debian GNU/Linux 10 
- **Modin version** (`modin.__version__`): 0.10.1
- **Python version**: 3.7.3
- **Code we can use to reproduce**: 

```
# works

import pandas as pd
df = pd.DataFrame({'a':[0],'b':[0]})
cols = ['a','b']
df[cols] = df[cols]

# does not work

import modin.pandas as pd
df = pd.DataFrame({'a':[0],'b':[0]})
cols = ['a','b']
df[cols] = df[cols]
```

### Describe the problem
With pd.DataFrame, assigning from a set of columns to a set of columns leads to the following exception with modin.pandas but not pandas. 

### Source code / logs
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/xxx/lib/python3.7/site-packages/modin/pandas/dataframe.py", line 2540, in __setitem__
    self._default_to_pandas(setitem_unhashable_key)._query_compiler
  File "/xxx/lib/python3.7/site-packages/modin/pandas/base.py", line 459, in _default_to_pandas
    result = op(pandas_obj, *args, **kwargs)
  File "/xxx/lib/python3.7/site-packages/modin/pandas/dataframe.py", line 2536, in setitem_unhashable_key
    df[key] = value
  File "/xxx/lib/python3.7/site-packages/pandas/core/frame.py", line 3597, in __setitem__
    self._setitem_array(key, value)
  File "/xxx/lib/python3.7/site-packages/pandas/core/frame.py", line 3648, in _setitem_array
    return self._setitem_array(key, value)
  File "/xxx/lib/python3.7/site-packages/pandas/core/frame.py", line 3643, in _setitem_array
    self._iset_not_inplace(key, value)
  File "/xxx/lib/python3.7/site-packages/pandas/core/frame.py", line 3670, in _iset_not_inplace
    raise ValueError("Columns must be same length as key")
ValueError: Columns must be same length as key

```

**Comment by devin-petersohn** [comment]
Closed by #3330

**Comment by JnsLns** [comment]
@YarShev @devin-petersohn 
Thanks a lot! Awesome!
